service/src/main/java/com/derpgroup/derpwizard/manager/AbstractManager.java

```diff
@@ -64,6 +82,31 @@ public void handleRequest(@NonNull VoiceInput voiceInput, @NonNull SsmlDocumentB
     }
   }
 
+  private void appendToConversationHistory(@NonNull VoiceInput voiceInput) {
+    CommonMetadata metadata = voiceInput.getMetadata();
+    List<ConversationHistoryEntry> conversationHistory = metadata.getConversationHistory();
+    if(conversationHistory == null){
+      conversationHistory = new ArrayList<ConversationHistoryEntry>();
+    }
+    CommonMetadata metadataClone = null;
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      mapper.registerModule(mapperModule);
+      String metadataString = mapper.writeValueAsString(metadata);
+      metadataClone = mapper.readValue(metadataString, CommonMetadata.class);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    
+    ConversationHistoryEntry entry = new ConversationHistoryEntry();
+    entry.setMessageMap(new LinkedHashMap<String,String>(voiceInput.getMessageAsMap()));
```

Comment:
Well, we're guaranteed that the JSON will serialize into that map in order - not sure if we're guaranteed to get the same order in the JSON from Alexa (but I think so).

And yes, we should maintain order.  As Rusty pointed out a couple weeks ago, there may be cases where all we have to work off of is the map of parameters, and order would be important in that case.